doc/nrf/doc_build_process.rst

```diff
@@ -0,0 +1,78 @@
+:file:`nrf/doc/CMakeLists.txt` contains the function :code:`add_docset` which is used to create targets for all the docsets.
+This includes the :code:`{docset}`/ :code:`{docset}-all`, :code:`{docset}-inventory`/ :code:`{docset}-inventory-all`, :code:`{docset}-linkcheck` and :code:`{docset}-clean` targets.
+The inventory targets create reference targets without building the documentation, providing a solution for any circular dependencies.
+Some docsets require additional source files to be generated before the docset can be built, like :ref:`Zephyr's device tree<devicetree-intro>`.
+In these cases a separate target is created and added as a dependency for the docset target.
+
+External contents
+-----------------
+When the :code:`ninja` command executes :code:`sphinx-build`, a :file:`_build/{docset}/src` directory is used as the source directory.
+An extension from the Zephyr docset :file:`external_content` executes before the build is initiated and copies every relevant documentation file into this folder.
+This is a "smart" copy, meaning that only new or modified files are copied to reduce build time.
+As this can change the file hierarchy somewhat, local link paths are modified during this process.
+Which files are copied is listed under :code:`external_content_contents` in the :file:`conf.py` file.
+
+Intersphinx
+-----------
+The |NCS| documentation consists of multiple docsets.
+The main docset is the NRF documentation, while the other docsets are pulled from their upstream versions.
+The only file that is needed to maintain for this, is a customized :file:`conf.py` file located under :file:`doc/{docset-name}/`.
+To be able to make references across docsets, the Sphinx extension Intersphinx is used, and local paths are configured under :code:`intersphinx_mapping` in :file:`doc/nrf/conf.py`.
+The upstream docsets are pulled using West.
+
+Building and publishing the documentation
+*****************************************
+The |NCS| documentation is currently hosted on :code:`developer.nordicsemi.com` under the folder :file:`nRF_Connect_SDK` and is updated with every pull request that is merged into the main branch of the Git repository.
+Every pull request made is also hosted under the :file:`nRF_Connect_SDK_dev` folder if the build workflow is successful.
+This is done through two chained GitHub actions, :file:`docbuild.yml` and :file:`docpublish.yml`.
+The steps in each of these workflows are listed below:
+
+1. Make PR, invoke GitHub action `docbuild`_
+
+    1. Checkout the code
+    2. Cache pip
+    3. Install packages
+    4. Install base dependencies
+    5. West init and update
+    6. Install documentation dependencies
+    7. Build the documentation (:code:`cmake` and :code:`ninja`)
+    8. Build cache with :file:`doc/_scripts/cache_create.py`, store in local folder :file:`cache`
+    9. Prepare extra cache files
+    10. Archive cache
```

Comment:
hmm I think this explains too many details, e.g. pip cache is not relevant here. I think it's better to explain that the workflow checks out all repositories, builds all docsets and generates a cache file which can then be used locally to speed up builds.